recipes/nv-codec-headers/all/conanfile.py

```diff
@@ -0,0 +1,52 @@
+from enum import auto
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class FFNvEncHeaders(ConanFile):
+    name = "nv-codec-headers"
+    description = "FFmpeg version of headers required to interface with Nvidia's codec APIs"
+    topics = ("ffmpeg", "video", "nvidia", "headers")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/FFmpeg/nv-codec-headers"
+    license = "MIT"
+
+    _autotools = None
+
+    def package_id(self):
+        self.info.header_only()
+
+    def build_requirements(self):
+        if tools.os_info.is_windows:
+            if "CONAN_MAKE_PROGRAM" not in os.environ and not tools.which("make"):
+                self.build_requires("make/4.2.1")
+
+    def source(self):
```

Comment:
This has the possibility to break on c3i.
Also, I am not sure about your test: `if "CONAN_MAKE_PROGRAM" not in os.environ and not tools.which("make")`.
`make` on Windows does provide `gnumake.exe`, not `make.exe`.


```suggestion
    _autotools = None

    @property
    def _settings_build(self):
        return self.settings_build if hasattr(self, "settings_build") else self.settings

    def build_requirements(self):
        if self._settings_build.os == "Windows":
            if not tools.get_env("CONAN_MAKE_PROGRAM") and not tools.which("make"):
                self.build_requires("make/4.2.1")

    def package_id(self):
        self.info.header_only()

    def source(self):
```